### PR TITLE
Set `app/packs` as default Webpacker folder instead of `app/javascript`

### PR DESCRIPTION
**Pull request by guillaumewrobel**
### Summary

The [current usage guide for Webpacker](https://github.com/rails/webpacker#usage) recommends to use `app/packs` for Webpacker files.

The difference between [Rails guides](https://edgeguides.rubyonrails.org/webpacker.html) and Webpacker README are confusing for developers. I suggest to apply the `app/packs` naming convention.

### Details

I replaced occurrences of `app/javascript` with `app/packs`.
Default `source_entry_path` should also be `entrypoints` and not `packs`.

### Comments
What do you think? What is gonna be the standard convention for Rails 7?

### CODE REVIEW

File: guides/source/webpacker.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Webpacker is installed by default in Rails 6.0 and up. You can install it with a
 
 |File                    |Location                |Explanation                                                                                         |
 |------------------------|------------------------|----------------------------------------------------------------------------------------------------|
-|JavaScript Folder       | `app/javascript`       |A place for your front-end source                                                                   |
+|JavaScript Folder       | `app/packs`       |A place for your front-end source                                                                   |
```
Comment (sunny) [inline guides/source/webpacker.md:58]:
```suggestion
|JavaScript Folder       | `app/packs`            |A place for your front-end source                                                                   |
```

File: railties/lib/rails/generators/rails/plugin/plugin_generator.rb [hunk 2]
```diff
@@ -151,7 +151,7 @@ def test_dummy_config
     end
 
     def test_dummy_webpacker_assets
-      template "rails/javascripts.js",    "#{dummy_path}/app/javascript/packs/application.js", force: true
+      template "rails/javascripts.js",    "#{dummy_path}/app/packs/entrypoints/application.js", force: true
```
Comment (sunny) [inline railties/lib/rails/generators/rails/plugin/plugin_generator.rb:154]:
```suggestion
      template "rails/javascripts.js", "#{dummy_path}/app/packs/entrypoints/application.js", force: true
```

File: railties/lib/rails/tasks/statistics.rake [hunk 3]
```diff
@@ -14,7 +14,7 @@ STATS_DIRECTORIES ||= [
   %w(Views              app/views),
   %w(JavaScripts        app/assets/javascripts),
   %w(Stylesheets        app/assets/stylesheets),
-  %w(JavaScript         app/javascript),
+  %w(JavaScript         app/packs),
```
Comment (sunny) [inline railties/lib/rails/tasks/statistics.rake:17]: Perhaps the name change should be reflected here as well?

```suggestion
  %w(Packs              app/packs),
```

**Comment by dhh** [comment]
Yeah, we're in a transition period. Webpacker 6 is not out yet, so Webpacker 5 is what Rails need to hit at the moment. And we're still not certain on what to do with Webpacker re: Rails 7 (around defaults etc). So think it's too early to update this.

**Pull request closed without merging**